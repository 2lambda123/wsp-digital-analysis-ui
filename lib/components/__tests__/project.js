/* global describe, it, expect, jest */

import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import React from 'react'

import Project from '../project'

describe('Component > Project', () => {
  it('renders correctly', () => {
    const loadFn = jest.fn()
    const tree = mount(
      <Project
        description='A test project'
        id='1234'
        isLoaded={false}
        load={loadFn}
        name='Test'
        >
        Project content
      </Project>
    )
    expect(mountToJson(tree)).toMatchSnapshot()
    expect(loadFn).toBeCalled()
  })
})
