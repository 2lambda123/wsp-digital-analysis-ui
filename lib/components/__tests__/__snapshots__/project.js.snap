// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > Project renders correctly 1`] = `
<div>
  <div
    className="DockTitle"
  >
    Project:
    <a
      onClick={[Function]}
      style={Object {}}
    >
      <span>
         
        Test
      </span>
    </a>
    <a
      className="pull-right"
      onClick={[Function]}
      style={Object {}}
    >
      <i
        className="fa fa-pencil fa-fw "
      />
    </a>
  </div>
</div>
`;

exports[`Component > Project renders unsupported message correctly 1`] = `
<div>
  <div
    className="DockTitle"
  >
    Project:
    <a
      onClick={[Function]}
      style={Object {}}
    >
      <span>
         
        Test
      </span>
    </a>
    <a
      className="pull-right"
      onClick={[Function]}
      style={Object {}}
    >
      <i
        className="fa fa-pencil fa-fw "
      />
    </a>
  </div>
  <div
    className="alert alert-danger"
    role="alert"
  >
    The chosen analysis backend version of this project is outdated.
     
    <a
      onClick={[Function]}
      style={Object {}}
    >
      Edit project.
    </a>
  </div>
</div>
`;
